cpp/src/glm/ols.cuh

```diff
@@ -91,15 +91,20 @@ void olsFit(const raft::handle_t& handle,
                    stream);
   }
 
-  if (algo == 0 || algo == 1) {
-    LinAlg::lstsq(handle, input, n_rows, n_cols, labels, coef, algo, stream);
-  } else if (algo == 2) {
-    LinAlg::lstsqQR(input, n_rows, n_cols, labels, coef, cusolver_handle, cublas_handle, stream);
-  } else if (algo == 3) {
-    ASSERT(false, "olsFit: no algorithm with this id has been implemented");
-  } else {
-    ASSERT(false, "olsFit: no algorithm with this id has been implemented");
+  int selectedAlgo = algo;
+  if (n_cols > n_rows || n_cols == 1) selectedAlgo = 0;
+
+  ML::PUSH_RANGE("Trace::MLCommon::LinAlg::ols-lstsq*", stream);
+  switch (selectedAlgo) {
+    case 0: LinAlg::lstsqSvdJacobi(handle, input, n_rows, n_cols, labels, coef, stream); break;
```

Comment:
Now that these primitives are beginning to get integrated into Spark ML, we should move these over to RAFT, since they plan to depend directly on RAFT and not cuML. Are you interested in doing the move?

Reply:
Do you mean moving `lstsq*` functions, or updating `raft::linalg::svd*` functions? i.e. add new primitives to solve OLS to raft, or just add one more SVD decomposition function (SVD Jacobi)?

Maybe I should first finish this PR within 21.10 as-is, and then do the refactoring in 21.12?

Reply:
@achirkin, I'm referring to specifically moving over the least squares solver wrapper prims to RAFT so that other libraries can also make use of them outside of cuml. There's no rush on this, so definitely not something that should hold up this PR. However, it would be nice to have them moved over soon (maybe even in 21.12 timeframe).

Reply:
Ok, why not!